gpg_group_chat/client/client.py

```diff
@@ -0,0 +1,73 @@
+import select
+import socket
+import sys
+import threading
+
+
+class Client():
+
+    def __init__(self):
+        self._socket = None
+        self._connected = False
+        self._nickname = 'anonymous'
+
+    def start(self, server_ip, server_port, nickname=None):
+        if nickname is not None:
+            self._nickname = nickname
+        print('Hello friend ' + self._nickname)
+        dest = (server_ip, server_port)
+        try:
+            self._socket = socket.create_connection(dest)
+        except:
+            print('error opening socket')
+            return
+        self.__handle_messages()
+
+    def __start_protocol(self):
+        msg = 'hello, i am ' + self._nickname + ',' + self._key
+        self._socket.sendall(msg.encode('utf-8'))
+        msg = self._socket.recv(1024)
+        if not msg:
+            print('no response from server')
+        else:
+            if(msg.startswith(b'ACK')):
+                print('the server accepted your connection')
+                print('the connected users are:')
+                print(msg[3:])
+            elif(msg.startswith(b'NACK')):
+                print('the server refused your connection')
+            else:
+                print('unknown answer from server, be careful')
+        return True
+
```

Comment:
I think that would be better move this protocol implementation to another PR, according with each open issue, this will make ssl/tls implementation easier.